stan/math/prim/scal/fun/init_threadpool_tbb.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef STAN_MATH_PRIM_SCAL_INIT_THREADPOOL_TBB_HPP
+#define STAN_MATH_PRIM_SCAL_INIT_THREADPOOL_TBB_HPP
+
+// TODO(SW): remove STAN_THREADS guards once Intel TBB is fully
+// mandatory
+
+#include <stan/math/prim/scal/fun/get_num_threads.hpp>
+
+#ifdef STAN_THREADS
+#include <tbb/task_scheduler_init.h>
+#endif
+
+namespace stan {
+namespace math {
+
+/**
+ * Initialize the Intel TBB threadpool through the
+ * tbb::task_scheduler_init object. In case an instance of the
+ * tbb::task_scheduler_object is instantiated, then this has no effect,
+ * which is the default behavior of the Intel TBB library.
+ *
+ * The maximal number of threads is read from the environment variable
+ * STAN_NUM_THREADS using internal::get_num_threads. See conventions
+ * of get_num_threads.
+ *
+ * The function returns as a boolean if the tbb::task_scheduler_init
+ * has become active once constructed. The instance may not become
+ * active if there is already another instance created before calling
+ * this function.
+ *
+ * @param stack_size sets the stack size of each thread; the default 0
+ * let's the TBB choose the stack size
+ * @return active status of static tbb::task_scheduler_init
+ * @throws std::runtime_error if the value of STAN_NUM_THREADS env. variable
+ * is invalid
+ */
+inline bool init_threadpool_tbb(tbb::stack_size_type stack_size = 0) {
```

Comment:
What happens in someone calls this more than once? Should this throw, return silently without changing anything? Is there a TBB API call you can make to check whether the threadpool was initialized?

Reply:
Is that what is_active does? Should we return immediately if is_active is true?

Reply:
Comment 1: First note that the `tbb::task_scheduler_init` is **static**. So you ever create a single instance of this thing. Any further call is basically ignored. This is in line with the TBB way of working in this case. Repeated initialisation is not possible. You do it once and that's it - no more reconfiguration after that. If you want to have differently sized working areas then you can use so-called `task_arenas`.

Comment 2: Why should we return immediately? In case the *static* init object is active, then this one was used (and the parsing has worked) so that we will just return again `true` if it was true before.